### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by 0xC0ncord**
More various fixes. Of note:
- Allow systemd-pcrphase to write TPM measurements in systemd 255
- Add a policy for systemd-machine-id-setup, which in systemd 255 now uses a tmpfs when committing the machine id
- Allow systemd to read the system DBUS daemon's anon pidfds (systemd 255)
- Fix CRI-O not starting due to lack of access to `/root/.config/containers`
- Fix Kubelet being unable to apply fsGroup ownership changes to unlabeled persistent volumes

### CODE REVIEW

**Review by pebenito — CHANGES_REQUESTED** [review]

File: policy/modules/system/systemd.te [hunk 1]
```diff
@@ -1097,6 +1102,30 @@ optional_policy(`
 	dbus_system_bus_client(systemd_machined_t)
 ')
 
+#########################################
+#
+# machine-id-setup local policy
+#
+
+allow systemd_machine_id_setup_t self:capability { setgid sys_admin sys_chroot };
```
[the remark this answers is not in the document]
Comment (0xC0ncord) [reply policy/modules/system/systemd.te:1110]: Correct.

File: policy/modules/system/systemd.te [hunk 2]
```diff
@@ -1477,6 +1506,12 @@ init_read_state(systemd_pcrphase_t)
 # for writing the TPM public key and measurements to /var/lib/systemd and /run/systemd
 init_manage_runtime_files(systemd_pcrphase_t)
 init_manage_var_lib_files(systemd_pcrphase_t)
+# for writing the TPM measurements to /var/log/systemd
+init_list_log_dirs(systemd_pcrphase_t)
+init_create_log_files(systemd_pcrphase_t)
+init_write_log_files(systemd_pcrphase_t)
+init_setattr_log_files(systemd_pcrphase_t)
+init_log_create_dirs(systemd_pcrphase_t)
```
[the remark this answers is not in the document]
Comment (0xC0ncord) [reply policy/modules/system/systemd.te:1514]: I haven't seen any other uses for it. How about `init_tpmlog_t`?

[the remark this answers is not in the document]
Comment (0xC0ncord) [reply policy/modules/system/systemd.te:1514]: There is the possibility that `/var/log/systemd` could gain additional purpose in the future, though.

[the remark this answers is not in the document]
Comment (pebenito) [reply policy/modules/system/systemd.te:1514]: Ok, we can keep log, but I think `systemd_log_t` would be better. Please move the comment up with the type decl.

[the remark this answers is not in the document]
Comment (pebenito) [reply policy/modules/system/systemd.te]: The type should go in the systemd module; I don't see a strong reason to keep it in `init.te`.

**Comment by pebenito** [comment]
Needs rebase.

**Review by pebenito — APPROVED** [review]

**Comment by 0xC0ncord** [comment]
> Needs rebase.

Had a mixup with branches during the last rebase. All good to go now. Thanks!

**Pull request merged**